sympy/external/gmpy.py

```diff
@@ -188,14 +168,32 @@ def iroot(x, n):
     MPQ = flint.fmpq # type: ignore
 
     factorial = python_factorial
-    sqrt = python_sqrt
-    is_square = python_is_square
-    sqrtrem = python_sqrtrem
-    gcd = python_gcd
-    lcm = python_lcm
+
```

Comment:
Do all these return Python types?

Reply:
If `GROUND_TYPES` is `'flint'`, are you asking if functions like `gcd` return `int`? If so, no.

When `GROUND_TYPES` is `'gmpy'`, these return values are `gmpy.mpz`. Similarly, the return value for `flint` will be `flint.fmpz`.